### PR TITLE
Add BigInt database field type with String representation in JS

### PR DESCRIPTION
**Pull request by MurzNN**
Here I added the BigInt field type to Keystone-Next, with support for all database adapter types. 

Requires also this patch for prisma adapter: https://github.com/keystonejs/keystone/pull/5007 if you need to use this field on relationships,or as id field. 

Code style is based on `packages-next/fields-document` implementation (without files in `@keystone-next/fields-legacy` package).

JavaScript and GraphQL have bad support for BigInt, so we store real BigInt values only in database, but represent them as strings in JavaScript and GraphQL schemas. GraphQL comparison operators accept strings too, but works like with number field type (with `_gt`, `_gte`, etc).

More info about BigInt support problems:
- in GraphQL https://github.com/graphql/graphql-js/issues/2017
- in JSON https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#use_within_json

P.S. Alternative realization can be with adding the [apollo-type-bigint](https://www.npmjs.com/package/apollo-type-bigint) package to Keystone's graphQLSchema, but I think this will not good to do as default, because of needs for support on client side too.

### CODE REVIEW

**Comment by MurzNN** [comment]
I rebased my PR to new core, most of new code is based on Integer and Decimal fields, please review it.

**Comment by MurzNN** [comment]
Until this field type is merged to Keystone core, I published it into https://github.com/keystonejs-contrib/keystonejs-contrib/tree/master/packages/fields-bigint for make it easier to include in other projects.

**Comment by dcousens** [comment]
Rebased on `main`

**Comment by dcousens** [comment]
Great work (again) on this @MurzNN!

Thinking about this today, we think the GraphQL schema for BigInt's should be a GraphQL scalar type,  so if you're OK with it we might make a few changes to this pull request as is,  and then merge that :+1: 

Again, great work and thanks for being patient on this.

**Comment by MurzNN** [comment]
Yeah, it's okay to make BigInt's a GraphQL scalar type! I'm really happy to see the progress with this feature, and will glad to have it in Keystone core! Cuz now we need to support our own fork to make it available for our projects.

**Comment by MurzNN** [comment]
@dcousens Also maybe you can also try to push a similar PR https://github.com/keystonejs/keystone/pull/7188 to allow using BigInt as list item id in Keystone?

**Comment by dcousens** [comment]
@mitchellhamilton why not use `**` in [19e86e6](https://github.com/keystonejs/keystone/pull/5175/commits/19e86e69d0a4c5b77195747c7a2ac2e0cf3f66bb)

**Comment by mitchellhamilton** [comment]
> why not use ** in 19e86e6?

Because with our current Babel config, it'll be compiled to `Math.pow` which only works on numbers not bigints and I didn't want to answer "what should our compile target be" in this PR

File: docs/pages/docs/apis/fields.mdx
Comment (dcousens) [inline docs/pages/docs/apis/fields.mdx:242]:
```suggestion
When interacting with the field with GraphQL, values will be strings.
```

**Review by dcousens — APPROVED** [review]
A few copy changes, but, I honestly don't know how I feel about this as a noun.
I think it is completely confusing to anyone who is moves between the domains of frontend and backend (our developers).

If we merge this pull request as is, we would have, as valid identifiers in Keystone

* `bigint` - **Typescript**, arbitrary precision integer
* `bigInt` - **Keystone**, field type for 64-bit wide integer
* `BigInt` - **Javascript**, arbitrary precision integer
* `BigInt` - **Keystone**, a GraphQL type, representing an arbitrary precision integer
* `BigIntFieldConfig` - **Keystone**, a Typescript type name for our 64-bit integer field type
* `BIGINT` - **POSTGRES**, 64-bit wide integer

I'm feeling like :exploding_head: - but I don't want to hold this back on a bike-shed.

File: docs/pages/docs/apis/fields.mdx
Comment (dcousens) [reply docs/pages/docs/apis/fields.mdx:247]: Are we typically using Javascript or Typescript terminology?

Comment (mitchellhamilton) [reply docs/pages/docs/apis/fields.mdx:247]: `bigint` _is_ the JavaScript terminology, it is a `bigint`, not a `BigInt` in the same way that there is `string` and `String` and you always want a `string`.

**Pull request merged**
